tasks/build/os_shell_scripts.js

```diff
@@ -5,37 +5,39 @@ import {ncp} from 'ncp';
 const pncp = promisify(ncp);
 const punlink = promisify(unlink);
 
-module.exports = function (grunt) {
+export default function (grunt) {
   grunt.registerTask('_build:osShellScripts', function () {
     const done = this.async();
     const source = 'build/kibana/bin';
     const platforms = grunt.config.get('platforms');
+    const allPlatforms = fn => invokeAllAsync(platforms, fn);
 
-    Promise.resolve()
-      .then(() => allPlatforms(platform => punlink(join(platform.buildDir, 'bin')))
-      .then(() => allPlatforms(platform => pncp(source, join(platform.buildDir, 'bin'))))
-      .then(() => allPlatforms(removeExtraneousShellScripts)))
-      .then(() => done)
-      .catch(err => done(err));
+    run()
+      .then(() => done())
+      .catch(e => done(e));
```

Comment:
Let me know if there's a `bluebird` method for this. Not that used to converting callbacks into promises